cookie.odin

```diff
@@ -77,7 +77,7 @@ cookie_write :: proc(w: io.Writer, c: Cookie) -> io.Error {
 }
 
 // Builds the Set-Cookie header string representation of the given cookie.
-cookie_string :: proc(c: Cookie, allocator := context.allocator) -> string {
+cookie_string :: proc(c: ^Cookie, allocator := context.allocator) -> string {
```

Comment:
Maybe I'm micro-optimizing :) but `cookie_string` was making its own copy only to pass to `cookie-write` which would make another copy.

Reply:
These aren't actually copied though, the Odin calling convention (and the fact that params are immutable) makes them already be passed by pointer. So this change shouldn't have actually changed the compiled output, it just makes it less clear imo (I usually only pass by pointer if needed so I can see what actually mutates it's parameters).

Reply:
Oh, I see. Is this documented somewhere?
I don't see it mentioned here: https://odin-lang.org/docs/overview/#parameters
or here: https://odin-lang.org/docs/overview/#calling-conventions
It just says `This allows for numerous optimizations with the Odin calling conventions`.

I wonder at which point it is decided whether they should be implicitly passed by pointer? Is there like a cutoff struct size that determines this?

Reply:
Ah, this is very good to know! Just saw your Discord comment on this from a few days ago as well.

Reply:
I was ready to merge the cookie struct packing though!

Reply:
Lol, let me undo that then.

Reply:
I force pushed, so had to create a new PR: https://github.com/laytan/odin-http/pull/58